PhysicsTools/NanoAOD/python/protons_cff.py

```diff
@@ -0,0 +1,74 @@
+from PhysicsTools.NanoAOD.common_cff import *
+from Configuration.Eras.Modifier_run2_miniAOD_80XLegacy_cff import run2_miniAOD_80XLegacy
+from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv1_cff import run2_nanoAOD_94XMiniAODv1
+from Configuration.Eras.Modifier_run2_nanoAOD_94XMiniAODv2_cff import run2_nanoAOD_94XMiniAODv2
+from Configuration.Eras.Modifier_run2_nanoAOD_94X2016_cff import run2_nanoAOD_94X2016
+from Configuration.Eras.Modifier_run2_nanoAOD_102Xv1_cff import run2_nanoAOD_102Xv1
+
+protonTable = cms.EDProducer("ProtonProducer",
+                             tagRecoProtonsSingle = cms.InputTag("ctppsProtons", "singleRP"),
+                             tagRecoProtonsMulti  = cms.InputTag("ctppsProtons", "multiRP"),
+                             tagTrackLite         = cms.InputTag("ctppsLocalTrackLiteProducer")
+)
+
+singleRPTable = cms.EDProducer("SimpleProtonTrackFlatTableProducer",
+    src = cms.InputTag("ctppsProtons","singleRP"),
+    cut = cms.string(""),
+    name = cms.string("Proton_singleRP"),
+    doc  = cms.string("bon"),
+    singleton = cms.bool(False),
+    extension = cms.bool(False),
+    skipNonExistingSrc = cms.bool(True),
+    variables = cms.PSet(
+        xi = Var("xi",float,doc="xi or dp/p",precision=12),
+        xiUnc = Var("xiError",float,doc="uncertainty on xi or dp/p",precision=8),
+        thetaY = Var("thetaY",float,doc="th y",precision=13),
+        thetaYUnc = Var("thetaYError",float,doc="theta Y uncertainty",precision=8),
+    ),
+    externalVariables = cms.PSet(
+        decRPId = ExtVar("protonTable:protonRPId",int,doc="Detector ID",precision=8), 
+        sector45 = ExtVar("protonTable:singleRPsector45",bool,doc="LHC sector 45"),
+        sector56 = ExtVar("protonTable:singleRPsector56",bool,doc="LHC sector 56"),
+    ),
+)
+
+multiRPTable = cms.EDProducer("SimpleProtonTrackFlatTableProducer",
+    src = cms.InputTag("ctppsProtons","multiRP"),
+    cut = cms.string(""),
+    name = cms.string("Proton_multiRP"),
+    doc  = cms.string("bon"),
+    singleton = cms.bool(False),
+    extension = cms.bool(False),
+    skipNonExistingSrc = cms.bool(True),
+    variables = cms.PSet(
+        xi = Var("xi",float,doc="xi or dp/p",precision=12),
+        xiUnc = Var("xiError",float,doc="uncertainty on xi or dp/p",precision=8),
+        vtxY = Var("vy()",float,doc="vertex y",precision=11),
+        vtxYUnc = Var("vyError",float,doc="vy Error",precision=8),
+        pt = Var("pt",float,doc="pt",precision=15),
+        thetaX = Var("thetaX",float,doc="theta x",precision=13),
+        thetaXUnc = Var("thetaXError",float,doc="theta X uncertainty",precision=8),
+        thetaY = Var("thetaY",float,doc="theta y",precision=13),
+        thetaYUnc = Var("thetaYError",float,doc="theta Y uncertainty",precision=8),
+        chi2 = Var("chi2",float,doc="chi 2",precision=15),
+        ndof = Var("ndof()",int, doc="n dof", precision=8),
+        t = Var("t",float,doc="t",precision=13),
```

Comment:
what is t ? not the time, since you have another variable for that, have the doc more verbose

Reply:
This is the Mandelstam variable "t". Not to be confused with time.